### PR TITLE
Added maximum number of receipts when creating payload

### PR DESCRIPTION
**Pull request by durkmurder**

### CODE REVIEW

File: module/builder/consensus/builder.go [hunk 1]
```diff
@@ -109,6 +113,10 @@ func (b *Builder) BuildOn(parentID flow.Identifier, setter func(*flow.Header) er
 		return nil, fmt.Errorf("could not insert receipts: %w", err)
 	}
 
+	if len(insertableReceipts) > DefaultPayloadMaximumReceipts {
+		insertableReceipts = insertableReceipts[:DefaultPayloadMaximumReceipts]
```
Comment (zhangchiqing) [inline module/builder/consensus/builder.go:117]: Nice 👍 

I would suggest to move this into `getInsertableReceipts`. The reason is that the insertableReceipts are ordered, adding it here would imply that we "know" about the order and how to pick the first X receipts. But I think we could let `getInsertableReceipts` to decide everything, including the order as well as what are the first X receipts:

```
	insertableReceipts, err := b.getInsertableReceipts(parentID, DefaultPayloadMaximumReceipts)
```

**Review by zhangchiqing — APPROVED** [review]
Looks good. Some minor comment

File: module/builder/consensus/builder.go (hunk 1 above)
Comment (durkmurder) [reply module/builder/consensus/builder.go:117]: I was also thinking about it, but it kinda makes sense to have a method which just returns all insertable seals and then filter it, looks kinda cleaner to me and might be useful.

**Review by arrivets — APPROVED** [review]

**Comment by AlexHentschel** [comment]
I am very sorry, this is probably on me: we have two PRs for the same work:
* https://github.com/onflow/flow-go/pull/292/files
* https://github.com/onflow/flow-go/pull/300/files

**Comment by AlexHentschel** [comment]
⚠️ 
I am very sorry, this is probably on me: we have two PRs for the same work:
* https://github.com/onflow/flow-go/pull/292/files
* https://github.com/onflow/flow-go/pull/300/files

My suggestion would be to go with https://github.com/onflow/flow-go/pull/300 as it includes a test case.

**Pull request closed without merging**